tests/lora/test_lora_hpu.py

```diff
@@ -0,0 +1,225 @@
+import pytest
+import torch
+
+from vllm.lora.layers import _apply_lora, _apply_lora_packed_nslice
+
+from .utils import DummyLoRAManager
+
+TENSOR_SIZES = [128, 1024, 2048, 4096, 8192, 11008, 11008 // 2, 11008 // 4]
+QKV_TENSOR_SIZES = [
+    (8192, 1024, 1024),
+    (8192 // 8, 1024 // 8, 1024 // 8),
+    (4096, 4096, 4096),
+    (4096 // 2, 4096 // 2, 4096 // 2),
+]
+BATCH_SIZES = [8, 32, 256]
+RANKS = [8]
+DTYPES = [torch.float16]
+TOLERANCES = {
+    torch.float16: (5e-3, 5e-3),
+    torch.bfloat16: (3e-2, 2e-2),
+}
+
+
+@pytest.mark.parametrize("m", TENSOR_SIZES)
+@pytest.mark.parametrize("n", TENSOR_SIZES)
+@pytest.mark.parametrize("k", BATCH_SIZES)
+@pytest.mark.parametrize("rank", RANKS)
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_apply_lora(m, n, k, rank, dtype) -> None:
+    manager = DummyLoRAManager()
+
+    module_name = "module"
+    weight = torch.rand([m, n], device="hpu", dtype=dtype)
+
+    manager.init_random_lora(module_name, weight, rank=rank)
+    lora = manager.get_module_lora(module_name)
+
+    input = torch.rand(k, n, device="hpu", dtype=dtype)
+    expected = input @ lora.lora_a @ lora.lora_b * lora.scaling
+
+    lora_a_stack = torch.zeros(8,
```

Comment:
We handle no LoRA case using zero valued A and B tensors. So need to append these matrices at the end. So the size will be max_loras + 1. Also please handle the for loop (line number 53) and indices ((line number 60) in the test case such that the last index of the LoRA matrices are untouched.

Reply:
I completely agree with your suggestion. However, while reviewing the original vllm repo for this task, I noticed that the test_lora.py file has been recently removed in [7ecee34](https://github.com/vllm-project/vllm/commit/7ecee3432110bae563c8756a66b54e5f08dc777d) and seems to have been replaced by other test files. Considering the upstream changes, what do you think about using the alternative test files instead?

Reply:
@JHLEE17 For now we are planning to add copies of original test with a "_hpu" suffix into HabanaAI/vllm-fork, therefore it should be ok to take this test into vllm-fork. Later when we upstream these changes to vllm-project we will figure out correct approach (maybe create a separate folder for hpu tests).